analyses/cms-open-data-ttbar/analysis.py

```diff
@@ -1,518 +1,361 @@
-
-    hs = ROOT.THStack("j4b1btag", "btag-variations ; H_{T} [GeV]")
-    for h, name in zip(freshstack, btag_variations):
-        print(name)
-        ptr = h.Rebin(2, name)
-        ptr.SetLineWidth(4)
-        ptr.SetTitle(name)
-        hs.Add(ptr)
-    hs.Draw("hist nostack plc")
-    c.Draw()
-    x = hs.GetXaxis()
-    x.SetRangeUser(120, 500)
-    x.SetTitleOffset(1.5)
-    x.CenterTitle()
-    c.BuildLegend(0.65, 0.7, 0.9, 0.9)
-    c.SaveAs("btag.png")
-
-    # Jet energy variations
-    jet_variations = ["nominal", "pt_scale_up", "pt_res_up"]
-    freshstack = analysisManager.GetVarStack(region="4j2b", variations=jet_variations)
-    hs = ROOT.THStack("4j2bjet", "Jet energy variations ; m_{bjj} [GeV]")
-    for h, name in zip(freshstack, jet_variations):
-        print(name)
-        h.SetFillColor(0)
-        h.SetLineWidth(4)
-        h.SetTitle(name)
-        hs.Add(h)
-    hs.Draw("hist nostack plc")
-    c.Draw()
-    x = hs.GetXaxis()
-    x.SetRangeUser(0, 600)
-    x.SetTitleOffset(1.5)
-    x.CenterTitle()
-    c.BuildLegend(0.65, 0.7, 0.9, 0.9)
-    c.SaveAs("jet.png")
-
-    # Save histograms to disk
-    with ROOT.TFile.Open(ARGS.histograms_output_file, "RECREATE") as output:
-        for process in analysisManager:
-            for variation in analysisManager[process]:
-                for region in analysisManager[process][variation]:
-                    hist_name = (
-                        f"{region}_{process}_{variation}" if variation != "nominal" else f"{region}_{process}"
-                    )
-                    hist = analysisManager[process][variation][region]
-                    if not isinstance(hist, ROOT.TH1D):
-                        hist = hist.GetValue()
-                    if hist.IsZombie():
-                        raise TypeError(hist_name)
-                    hist_sliced = ROOT.Slice(hist, 120, 550)
-                    hist_binned = hist_sliced.Rebin(2, hist.GetTitle())
-                    output.WriteObject(hist_binned, hist_name)
-
-
-def main():
-
-    if ARGS.scheduling_mode == "imt":
-        init_functions()
-        ROOT.EnableImplicitMT(ARGS.ncores)
-        print(f"The num of threads = {ROOT.GetThreadPoolSize()}")
-        # No handle needed in local mode
-        connection = None
+            varied_histos = variationsfor_func(nominal_histo)
+            results.append(AGCResult(varied_histos, region, process, variation, nominal_histo))
+        else:
+            results.append(AGCResult(nominal_histo, region, process, variation, nominal_histo))
+        print(f"Booked histogram {histo_model.fName}")
+
+    # Return the booked results
+    # Note that no event loop has run yet at this point (RDataFrame is lazy)
+    return results
+
+
+def load_cpp():
+    """Load C++ helper functions. Works for both local and distributed execution."""
+    try:
+        localdir = get_worker().local_directory
```

Comment:
Maybe add comment
```
# Distributed RDataFrame will send the 'helpers.cpp' file to the local directory of every Dask worker
```

Reply:
thanks, added a comment also pointing to where that happens :+1: